### ISSUE TITLE
Mix dependency :only environment conflict in umbrella app

### ISSUE THREAD

**Issue by hunterboerner** [elixir-lang/elixir#3820]
The error:

```
# .../umbrella/apps/b 
$ iex -S mix
Unchecked dependencies for environment dev:
* excoveralls (Hex package)
  the dependency excoveralls

  > In mix.exs:
    {:excoveralls, "~> 0.3.0", [hex: :excoveralls, only: [:test, :dev]]}

  does not match the environments calculated for

  > In .../umbrella/apps/a/mix.exs:
    {:excoveralls, "~> 0.3.0", [hex: :excoveralls]}

  Remove the :only restriction from your dep
** (Mix) Can't continue due to errors on dependencies
```

App B depends on App A with `{:a, in_umbrella: true}`. App A and app B both include `{:excoveralls, "~> 0.3.0", only: [:test, :dev]}` in their deps.

I'm on Elixir 1.1.0 and I don't remember having this problem on 1.0.5. Is that enough to reproduce?

/cc @ericmj

**Comment by josevalim** [comment]
Please try elixir v1.1.1 or the v1.1 branch. I believe it has been fixed. :)

-- 


*José Valim*
www.plataformatec.com.br
Skype: jv.ptec
Founder and Director of R&D

**Comment by hunterboerner** [comment]
I upgraded to Elixir 1.1.1 (from Homebrew if that matters) and that doesn't seem to have helped.

**Comment by josevalim** [comment]
Thank you! I have verified and the fix exists in the v1.1 branch only: https://github.com/elixir-lang/elixir/commit/eb8d293ac5a9b9072b85714101c80ecdd507e6b4

Can you please give it a try? Thank you!

**Comment by hunterboerner** [comment]
Yep, `v1.1` branch works. I guess I'll just use that branch until that commit gets into a release.
